src/browser/fetch/response.rs

```diff
@@ -92,6 +92,35 @@ impl Response {
         }
     }
 
+    /// Check that response status is ok (2xx), in case of error convert the error response
+    /// to a provided type T, that has to implement `FromStr` trait.
+    ///
+    /// ```rust
+    /// fetch(url).await?.check_detailed_status::<String>()?
+    ///
+    /// ```
+    ///
+    /// # Errors
+    /// Returns a tuple of `FetchError` and `Option<T>` if status isn't 2xx.
+    pub async fn check_detailed_status<T>(
+        self,
+    ) -> std::result::Result<Self, (FetchError, Option<T>)>
+    where
+        T: std::str::FromStr,
+    {
+        let status = self.status();
+        if status.is_ok() {
+            Ok(self)
+        } else {
+            let value = if let Ok(text) = self.text().await {
+                T::from_str(&text).ok()
```

Comment:
How about returning the raw text and leaving it up to the user to decide how to process the data?

Reply:
Huh, that's a very fair point. It'll simplify the code and make it more useful (as the potential `ParseError` from `from_str()` won't be lost). I'll change it.